### DISCUSSION TITLE
3D models with animation do not work properly.

### DISCUSSION THREAD

**Discussion by yukipioa** [google/model-viewer#1853] [Q&A]
First of all, I apologize that my English is not clear.

3D models with animation do not work properly.
I have tried three models so far.

In the first one, the animation worked fine in preview mode (the first web screen that appears), but not in AR mode.
The file formats at this time are glb and usdz.

In the second case, the 3D model was not displayed in preview mode, but in AR mode, the 3D model was displayed and the animation worked correctly.
In this case, the file formats are gltf and usdz.

Third, the animation worked fine in preview mode, but not in AR mode.
The file format at this time is glb, usdz.

Based on the above, I hypothesized that the behavior would be different depending on the file format, so I changed the file format of the third 3D model to gltf, but the result was the same: the animation worked fine in preview mode, but not in AR mode.

I believe that there is a more fundamental cause than the file format, but I don't know what it is.
If anyone knows, I would appreciate it if you could let me know.

I am using the demo version of the HTML.
The HTML description is as follows

``` html
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Monkey</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" href="demo-styles.css">
    
    <!-- The following libraries and polyfills are recommended to maximize browser support -->
    <!-- NOTE: you must adjust the paths as appropriate for your project -->
    
    <!-- 🚨 REQUIRED: Web Components polyfill to support Edge and Firefox < 63 -->
    <script src="https://unpkg.com/@webcomponents/webcomponentsjs@2.1.3/webcomponents-loader.js"></script>

    <!-- 💁 OPTIONAL: Intersection Observer polyfill for better performance in Safari and IE11 -->
    <script src="https://unpkg.com/intersection-observer@0.5.1/intersection-observer.js"></script>

    <!-- 💁 OPTIONAL: Resize Observer polyfill improves resize behavior in non-Chrome browsers -->
    <script src="https://unpkg.com/resize-observer-polyfill@1.5.0/dist/ResizeObserver.js"></script>
    
    <!-- 💁 OPTIONAL: The :focus-visible polyfill removes the focus ring for some input types -->
    <script src="https://unpkg.com/focus-visible@5.0.2/dist/focus-visible.js" defer></script>

  </head> 
<body>
  <div id="card">
    <!-- All you need to put beautiful, interactive 3D content on your site: -->
    <model-viewer camera-controls autoplay
                  ar
                  shadow-intensity="1"
                  src="monkey.gltf"
                  ios-src="monkey.usdz"
                  alt="A 3D model of an monkey">
    </model-viewer>    
  </div>
    
  <!-- 💁 Include both scripts below to support all browsers! -->

  <!-- Loads <model-viewer> for modern browsers: -->
  <script type="module"
      src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js">
  </script>

  <!-- Loads <model-viewer> for old browsers like IE11: -->
  <script nomodule
      src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js">
  </script>
</body>
</html>
```

**Comment by elalish** [comment]
Given that you're talking about usdz, I assume you're testing on an iPhone? It's helpful to put the OS and browser versions so we know what you're testing. AR mode on iOS is QuickLook, which is a completely separate native app reading a different file, so any issues there will be unrelated to issues you have on the web. You might try the validator for glTF/GLB problems: https://github.khronos.org/glTF-Validator/

**Reply by yukipioa** [reply]
Thank you for your reply.
I apologize for the lack of information.
And thank you for letting me know.

I used my iPhone and opened it in Safari.

After reading your reply, I used android and opened it in chrome to verify, and the first and third 3D models were working fine.
The second 3D model turned out to be a problem with the object from the URL you provided.
Thank you very much.

Does this mean that the usdz 3D model is the cause of the problem?

**Reply by elalish** [reply]
Yes, that seems likely. I'm also not sure how QuickLook decides to trigger an animation. Check their documentation; they have some URL parameters you can put on `ios_src` to customize it a bit.

**Comment by gamemakers9** [comment]
WebCL has almost all what it needs to support like vulcan glb / gltf 2 in webassembly (Gltf SceneViewer wasm.js support more hardware?)
